src/features/logging/aggregate/index.js

```diff
@@ -15,49 +20,87 @@ export class Aggregate extends AggregateBase {
-        message,
-        attributes,
-        level
-      )
+    if (this.blocked) return
+    const log = new Log(
+      this.#agentRuntime.timeKeeper.convertRelativeTimestamp(timestamp),
+      message,
+      attributes,
+      level
     )
+    const logBytes = log.message.length + stringify(log.attributes).length + log.logType.length + 10 // timestamp == 10 chars
+    if (logBytes > MAX_PAYLOAD_SIZE) {
+      handle(SUPPORTABILITY_METRIC_CHANNEL, ['Logging/Harvest/Failed/Seen', logBytes])
+      return warn(LOGGING_IGNORED + '> ' + MAX_PAYLOAD_SIZE + ' bytes', log.message.slice(0, 25) + '...')
+    }
+
+    if (this.estimatedBytes + logBytes >= MAX_PAYLOAD_SIZE) {
+      handle(SUPPORTABILITY_METRIC_CHANNEL, ['Logging/Harvest/Early/Seen', this.estimatedBytes + logBytes])
+      this.scheduler.runHarvest({})
+    }
+    this.estimatedBytes += logBytes
+    this.bufferedLogs.push(log)
   }
 
-  preparePayload () {
+  prepareHarvest () {
+    if (this.blocked || !(this.bufferedLogs.length || this.outgoingLogs.length)) return
+    /** populate outgoing array while also clearing main buffer */
     this.outgoingLogs.push(...this.bufferedLogs.splice(0))
+    this.estimatedBytes = 0
+    /** see https://source.datanerd.us/agents/rum-specs/blob/main/browser/Log for logging spec */
     return {
-      common: {
-        attributes: {
-          entityGuid: this.#agentRuntime.appMetadata?.agents?.[0]?.entityGuid,
-          session: {
-            id: this.#agentRuntime?.session?.state.value || '0', // The session ID that we generate and keep across page loads
+      qs: {
+        browser_monitoring_key: this.#agentInfo.licenseKey
+      },
+      body: {
+        common: {
+          /** Attributes in the `common` section are added to `all` logs generated in the payload */
+          attributes: {
+            entityGuid: this.#agentRuntime.appMetadata?.agents?.[0]?.entityGuid, // browser entity guid as provided from RUM response
+            session: this.#agentRuntime?.session?.state.value || '0', // The session ID that we generate and keep across page loads
             hasReplay: this.#agentRuntime?.session?.state.sessionReplayMode === 1, // True if a session replay recording is running
             hasTrace: this.#agentRuntime?.session?.state.sessionTraceMode === 1, // True if a session trace recording is running
-            pageTraceId: this.#agentRuntime.ptid // The page's trace ID (equiv to agent id)
-          },
-          agent: {
-            appId: this.#agentInfo.applicationID, // Application ID from info object
-            standalone: this.#agentInfo.sa, // Whether the app is C+P or APM injected
-            version: this.#agentRuntime.version, // the browser agent version
-            distribution: this.#agentRuntime.distMethod // How is the agent being loaded on the page
+            ptid: this.#agentRuntime.ptid, // page trace id
+            appId: this.#agentInfo.applicationID, // Application ID from info object,
```

Comment:
`app_id` is what SR uses

Reply:
nvm, Blob consumer qs.app_id gets remapped (?)

Reply:
confirmed, remapped to `appId`